kanidmd/src/lib/value.rs

```diff
@@ -210,6 +215,7 @@ impl SyntaxType {
             SyntaxType::Utf8StringIname => 14,
             SyntaxType::NsUniqueId => 15,
             SyntaxType::DateTime => 16,
+            SyntaxType::EmailAddress => 17,
```

Comment:
Yes, there is a reason to go back and forth. Do I remember why? No, not at all. 

I think I hardcoded the usize here so that if the fields are re-arranged, we keep the values stable. More investigation needed?

Reply:
Is the `SyntaxType` used to describe a type of message being sent back and forth, I have a hard time imagining we would want to change it back and forth on the same machine. If this is the case, couldn't we use some [`serde` macros](https://serde.rs/variant-attrs.html) on the `SyntaxType` variants to tell each one to serialize to a unique `usize`? This would let us have associated `usize`s in the definition, instead of having different functions containing massive `match` statements.

Something like this:
```rust
pub enum SyntaxType {
    #[serde(rename = "0")]
    UTF8STRING,
    #[serde(rename = "1")]
    Utf8StringInsensitive,
    ...
    #[serde(rename = "16")]
    DateTime,
    #[serde(rename = "17")]
    EmailAddress,
}
```

I'm not sure if it will allow serialization to `usize`s though, or only strings.

Reply:
I think this is a case for a seperate issue to investigate and clean up to be sure of what we are doing.

Reply:
@QnnOkabayashi did you want to open this issue?